niri-ipc/src/lib.rs

```diff
@@ -1,19 +1,35 @@
 //! Types for communicating with niri via IPC.
 //!
-//! After connecting to the niri socket, you can send a single [`Request`] and receive a single
-//! [`Reply`], which is a `Result` wrapping a [`Response`]. If you requested an event stream, you
-//! can keep reading [`Event`]s from the socket after the response.
+//! After connecting to the niri socket, you can send one or multiple [`Request`]s as
+//! newline-separated JSON objects, to each of which niri responds to with a (also JSON-encoded and
+//! newline-separated) [`Reply`], which is a [`Result`] wrapping a [`Response`].
 //!
-//! You can use the [`socket::Socket`] helper if you're fine with blocking communication. However,
-//! it is a fairly simple helper, so if you need async, or if you're using a different language,
-//! you are encouraged to communicate with the socket manually.
+//! <div class="warning">
+//!
+//! Note that even if you send multiple [`Request`]s at once, the `niri` event loop might still
+//! interupt the execution of these messages between executing the current and the next [`Request`].
+//! That means that if you send multiple messages at once, the compositor state might change, which
+//! could cause unexpected results.
+//!
```

Comment:
When #1017 is implemented, something like

```suggestion
//!
//! If you **need** to ensure the atomicity of a list of [`Request`]s, use `<link-to-batching-action>`.
//!
```

should be added here.